### ISSUE TITLE
No Audio 32 bit VSTi

### ISSUE THREAD

**Issue by tank-trax** [LMMS/lmms#4911]
Debian 9.8 LMMS 1.2.0-rc8.427

LMMS loads 64 bit VST with audio
LMMS loads 32 bit VST with no audio

**Comment by tank-trax** [comment]
I have build two different versions using two methods:

one using 

`git clone --recurse-submodules -b master https://github.com/lmms/lmms`

the other using 

```
git clone https://github.com/LMMS/lmms.git
cd lmms/
git submodule update --init --recursive
```

the former (--recurse-submodules) will build both Vestige 32 and 64 bit

/lib/lmms/RemoteVstPlugin64
/lib/lmms/RemoteVstPlugin64.exe.so
/lib/lmms/32/RemoteVstPlugin32
/lib/lmms/32/RemoteVstPlugin32.exe.so

however only the 64 bit produces audio
the 32 bit meters move but no sound

this build is also missing these LADSPA plugs:
/lib/lmms/ladspa: calf.so
/lib/lmms/ladspa: ladspa-util.so
/lib/lmms: libOPL2.so
/lib/lmms: libpapu.so

====================================

the latter method (submodule update --init --recursive)
will only build the 32 bit Vestige

/lib/lmms/RemoteVstPlugin
/lib/lmms/RemoteVstPlugin.exe.so

audio is good with 32 bit VSTi's

this build is not missing these LADSPA plugs:
/lib/lmms/ladspa: calf.so
/lib/lmms/ladspa: ladspa-util.so
/lib/lmms: libOPL2.so
/lib/lmms: libpapu.so

**Comment by damian01w** [comment]
Please, check if your issue is related to #4846

**Comment by tank-trax** [comment]
I don't use the AppImage because it seg faults on my system

```
./lmms-1.2.0-rc7-linux-x86_64.AppImage 
Carla appears to be installed on this system at /usr/lib[64]/carla so we'll use it.
/tmp/.mount_lmms-1tNASDA/AppRun: line 17: ldconfig: command not found
Jack does not appear to be installed.  That's OK, we'll use a dummy version instead.
/tmp/.mount_lmms-1tNASDA/usr/bin/lmms.real: /tmp/.mount_lmms-1tNASDA/usr/lib/libdbus-1.so.3: no version information available (required by /usr/lib/x86_64-linux-gnu/libjack.so.0)
convolution: samplerate mismatch preset:48000 host:44100
convolution: samplerate mismatch preset:48000 host:44100
convolution: samplerate mismatch preset:48000 host:44100
convolution: samplerate mismatch preset:48000 host:44100
convolution: samplerate mismatch preset:48000 host:44100
convolution: samplerate mismatch preset:48000 host:44100
/tmp/.mount_lmms-1tNASDA/AppRun: line 23: 12298 Segmentation fault      QT_X11_NO_NATIVE_MENUBAR=1 $DIR/usr/bin/lmms.real "$@"

```

**Comment by tank-trax** [comment]
I think that is the reason why I decided to build it rather than use the AppImage 

I tried the one from the link but it was worse. It looks like config files are causing confusion. Even if I delete ~/lmms the seg faults persist

```
./lmms-1.2.0-rc8.8-linux-x86_64.AppImage 
Carla appears to be installed on this system at /usr/lib[64]/carla so we'll use it.
Jack appears to be installed on this system, so we'll use it.
/tmp/.mount_lmms-11JbtmN/usr/bin/lmms.real: /tmp/.mount_lmms-11JbtmN/usr/lib/libdbus-1.so.3: no version information available (required by /usr/lib/x86_64-linux-gnu/libjack.so.0)
VstSyncController: shmget() failed

/tmp/.mount_lmms-11JbtmN/AppRun: line 24: 12325 Segmentation fault      QT_X11_NO_NATIVE_MENUBAR=1 $DIR/usr/bin/lmms.real "$@"
```

**Comment by tank-trax** [comment]
This issue is not related to #4846

The builds I have made work fine. Audio is OK. 

I can use JACK. However, I can't use ALSA as the Audio Interface does not allow for a jack or default option and only displays the PulseAudio info of the soundcard, likely because I am using USB audio. Many Linux apps behave this way. Basically saying it is is being used. Any app that has default or jack though will connect. For example Renoise and Bitwig can't and Waveform can. The old version of LMMS could too.

I can also use the PortAudio Audio Interface as there is an option in its sub-settings for ALSA that does recognize jack and default.

Just to reiterate, the issue I am submitting is pertaining to Vestige. 

- if I build using 
`git clone --recurse-submodules -b master https://github.com/lmms/lmms`
it will build a Vestige that can load both 64 and 32 bit DLL files but only the 64 bit will produce audio

- if I build using `submodule update --init --recursive`
it will build a Vestige that can only load 32 bit DLL files and will produce audio

**Comment by tank-trax** [comment]
I get this error during compile - not sure if it is of consequence 

AutoMoc warning
---------------
"~/src/lmms/src/tracks/InstrumentTrack.cpp"
The file includes the moc file "InstrumentTrack.moc", but does not contain a Q_OBJECT, Q_GADGET or Q_NAMESPACE macro.

**Comment by tank-trax** [comment]
I guess you can close this ticket :: I did a git pull and saw that there were a few changes. Rebuilt from scratch and now both 64 and 32 bit DLL's load and produce audio :smile: 

but is still missing those LADSPA plugins required by a few of the demos

/lib/lmms/ladspa: calf.so
/lib/lmms/ladspa: ladspa-util.so
/lib/lmms: libOPL2.so
/lib/lmms: libpapu.so

**Comment by SecondFlight** [comment]
Hey, glad your issue is resolved!

I don't have the experience to comment on VST support or on missing plugins, but I'm also not sure who to tag. Either way, that's out of the scope of this issue, so I think I'll close it for now. Feel free to open a new issue about that (assuming there isn't one already).

**Comment by DomClark** [comment]
I think the VST audio issue was probably a variation of #4861, fixed by 3aeacca7acf7fe77a3f0545451dbe4e3332c1628. As for missing plugins, some of them have been renamed on master but the sample projects haven't been updated yet to reflect this. `calf` has changed to `veal`, `OPL2` to `opulenz`, and `papu` to `freeboy`. I'm not sure where `ladspa-util` comes from.
